### PR TITLE
MM-48688 Add separate benchmark flow for master/release branches

### PR DESCRIPTION
**Pull request by hmhealey**
When the previous version of this downloaded the web app and server for running benchmarks, it tried downloading a server+webapp bundle from pr-builds.mattermost.com, but that's only available for PR builds and not for the master/release branches. pr-builds does still have the web app code for those branches though (uploaded by the upload-s3 step of the CircleCI flow), so we can use that plus a server build from releases.mattermost.com combined together to run these benchmarks on.

This PR obviously uses the PR flow so it won't exercise the master/release flow, but an earlier version of this pretended to be on master, and you can see that run [here](https://github.com/mattermost/mattermost-webapp/actions/runs/3603560912/jobs/6071906015) where the download step runs through the master/release flow.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48688

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/performance-benchmarks.yml [hunk 1]
```diff
@@ -47,9 +47,25 @@ jobs:
       - name: Download the thing
         working-directory: e2e/cypress
         run: |
-          curl https://pr-builds.mattermost.com/mattermost-webapp/commit/${{ github.event.pull_request.head.sha || github.sha }}/mattermost-enterprise-linux-amd64.tar.gz > mattermost.tar.gz
-          tar -xvf mattermost.tar.gz
+          if [[ "${{ github.event.pull_request.head.sha }}" != "" ]]; then
+            echo "Using server build from pr-builds.mattermost.com"
+            curl https://pr-builds.mattermost.com/mattermost-webapp/commit/${{ github.event.pull_request.head.sha }}/mattermost-enterprise-linux-amd64.tar.gz > mattermost.tar.gz
 
+            tar xf mattermost.tar.gz
```
Comment (larkox) [inline .github/workflows/performance-benchmarks.yml:54]: Probably nothing, but I noticed you used to call `tar -xvf` and now you are calling `tar xf`.

**Review by larkox — APPROVED** [review]
Still not my forte this part of the code, but LGTM.

File: .github/workflows/performance-benchmarks.yml (hunk 1 above)
Comment (hmhealey) [reply .github/workflows/performance-benchmarks.yml:54]: Yeah, that was intentional. I wanted to simplify the output and match what the server calls here. I was kind of surprised to find that the `-` was unnecessary though

**Comment by amyblais** [comment]
@hmhealey Does this need to be cherry-picked to the ``release-7.6`` branch?

**Comment by hmhealey** [comment]
@amyblais No, it doesn't need this change since these benchmark tests never made it onto that branch to begin with

**Pull request merged**